### ISSUE TITLE
Return doesn't work properly with if-else in match

### ISSUE THREAD

**Issue by Chuhaa** [ballerina-platform/ballerina-lang#26798]
**Description:**
 $title
```
error: this function must return a result
```

**Steps to reproduce:**
```
function findAnimal(string animal) returns string{
    int total = 1;
    match animal {
        "Mouse" => {
            if (total == 2) {
                return "Total 2";
            }
            else {
                return "Not 2";
            }    
        }
        _ => {
            return "!Mouse";
        }
    }
}
```

**Affected Versions:**
slp5

**Comment by MaryamZi** [comment]
Doing the following (removing the else block and directly returning) works, but the original example should also work.

```ballerina
function findAnimal(string animal) returns string {
    int total = 1;
    match animal {
        "Mouse" => {
            if (total == 2) {
                return "Total 2";
            }
            return "Not 2";
        }
        _ => {
            return "!Mouse";
        }
    }
}
```

**Comment by KavinduZoysa** [comment]
Fixed in #27343
